integrations/redis.go

```diff
@@ -0,0 +1,154 @@
+package integrations
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-redis/redis/v7"
+	"github.com/reddit/baseplate.go/metricsbp"
+	"github.com/reddit/baseplate.go/tracing"
+)
+
+const (
+	fail    = "fail"
+	success = "success"
+)
+
+type redisContextKey int
+
+const (
+	redisCommandName redisContextKey = iota
+	redisCommandStartTime
+)
+
+func setRedisCommandStartTime(ctx context.Context) context.Context {
+	return context.WithValue(ctx, redisCommandStartTime, time.Now().UTC())
+}
+
+func getRedisCommandStartTime(ctx context.Context) (start time.Time) {
+	if t := ctx.Value(redisCommandStartTime); t != nil {
+		if startTime, ok := t.(time.Time); ok {
```

Comment:
Also nit, I think the nil check could be omitted here. `nil.(time.Time)` will just return (zero value, false) so the ok check should be sufficient. (for types that could be nil, for example `*time.Time`, it's possible that it's actually `nil` with type `*time.Time` so the type cast would return (nil, true), but since there's no code to set it that way we won't need to worry about that case).